From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Samuel Attard <samuel.r.attard@gmail.com>
Date: Tue, 30 Jun 2020 15:38:50 -0700
Subject: fix: ensure that self is retained until the RACSignal is complete

Looks like the clang that Chromium uses is slightly smarter with ARC than whatever Squirrel was built with previously.  We now need to keep a reference to self to keep it alive into the "then" of the RACSignal.

diff --git a/Squirrel/SQRLZipArchiver.m b/Squirrel/SQRLZipArchiver.m
index 478509cdd528db4fcfa340c6f93fa58a446957e6..7c279bf73c368453bff4f922d76908c06dc378cd 100644
--- a/Squirrel/SQRLZipArchiver.m
+++ b/Squirrel/SQRLZipArchiver.m
@@ -134,7 +134,7 @@ - (RACSignal *)launchWithArguments:(NSArray *)arguments {
 			return [RACSignal
 				zip:@[ self.taskTerminated, self.standardErrorData ]
 				reduce:^(NSNumber *exitStatus, NSData *errorData) {
-					if (exitStatus.intValue == 0) return [RACSignal empty];
+					if (exitStatus.intValue == 0) return [RACSignal return:self];
 
 					NSMutableDictionary *userInfo = [NSMutableDictionary dictionary];
 					userInfo[SQRLZipArchiverExitCodeErrorKey] = exitStatus;
